src/widgets/modals/ByProgramModal.js

```diff
@@ -0,0 +1,406 @@
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { StyleSheet, View } from 'react-native';
+
+import { PageContentModal } from './PageContentModal';
+
+import globalStyles, { DARK_GREY, WARM_GREY, SUSSOL_ORANGE } from '../../globalStyles';
+import { AutocompleteSelector, ToggleBar, PageButton, TextEditor } from '..';
+
+import { SETTINGS_KEYS } from '../../settings';
+import { getAllPrograms, getAllPeriodsForProgram } from '../../utilities/byProgram';
+import SequentialSteps from '../SequentialSteps';
+
+const localization = {
+  program: 'program',
+  supplier: 'supplier',
+  orderType: 'order type',
+  period: 'period',
+  programTitle: 'Select a Program to use',
+  supplierTitle: 'Select a Supplier to use',
+  orderTypeTitle: 'Select a Order Type to use',
+  periodTitle: 'Select a Period to use',
+  nameTitle: 'Give your stocktake a name',
+  programError: 'No programs available',
+  supplierError: 'No suppliers available',
+  orderTypeError: 'No order types available',
+  periodError: 'No periods available',
+};
+
+const getNewState = command => {
+  let newState;
+  switch (command) {
+    case 'RESET_ALL':
+      newState = {
+        program: {},
+        supplier: {},
+        period: {},
+        periods: null,
+        orderType: {},
+        orderTypes: null,
+        name: null,
+        currentStep: 0,
+      };
+      break;
+    case 'SELECT_PROGRAM':
+      newState = {
+        period: {},
+        periods: null,
+        supplier: {},
+        orderType: {},
+        orderTypes: null,
+        modalIsOpen: false,
+        currentStep: 1,
+      };
+      break;
+    case 'SELECT_ORDER_TYPE':
+      newState = { period: {}, periods: null, modalIsOpen: false };
+      break;
+    case 'SELECT_NAME':
```

Comment:
what's the difference between name and supplier ?

Reply:
supplier is the name object - I think I should have named this `otherStoreName` to be more consistent with mobile. Name is the name of a stock take

Reply:
Actually noticed supplier is used a lot also throughout the app. I think `otherStoreName` is actually kind of a bad variable name, so sticking with `supplier` unless there are objections